src/Init/Prelude.lean

```diff
@@ -2373,6 +2373,12 @@ Codepoint positions (counting the Unicode codepoints rather than bytes)
 are represented by plain `Nat`s instead.
 Indexing a `String` by a byte position is constant-time, while codepoint
 positions need to be translated internally to byte positions in linear-time.
+
+A byte position `p` is *valid* for a string `s` if `0 ≤ p ≤ s.endPos` and `p`
+lies on a UTF8 byte boundary. This notion is properly introduced in [std's
+`String.Pos.Valid`][valid].
+
+[valid]: https://leanprover-community.github.io/mathlib4_docs/Std/Data/String/Lemmas.html#String.Pos.Valid
```

Comment:
```suggestion
lies on a UTF8 byte boundary.
```
We shouldn't be linking to external URLs like this before we get a way to keep them working over time, even if it would be nice.

Reply:
Idem ditto. We've been using permalinks on zulip for years.